lib/model/query/datasets.js

```diff
@@ -123,7 +131,25 @@ const createOrMerge = (parsedDataset, form, fields) => async ({ one, Actees, Dat
 };
 
 // Insert a Dataset when there is no associated form and immediately publish
-const createPublishedDataset = (dataset, project) => async ({ one, Actees }) => {
+const createPublishedDataset = (dataset, project) => async ({ one, Actees, Datasets }) => {
+
+  // Check for existing dataset with the same name
+  const existingDataset = await Datasets.get(project.id, dataset.name, false);
+  if (existingDataset.isDefined()) {
+    const { publishedAt, id: datasetId } = existingDataset.get();
+    if (publishedAt)
+      throw Problem.user.uniquenessViolation({ table: 'datasets', fields: [ 'name', 'projectId' ], values: [ dataset.name, project.id ], });
+
+    // If existing dataset is only a draft, allow it to be published
+    return new Dataset(await one(sql`UPDATE datasets SET "publishedAt" = "createdAt" where id = ${datasetId} RETURNING *`));
+  }
+
+  // Check for a published dataset with a similar name but different capitalization
+  const conflictingName = await Datasets.getPublishedBySimilarName(project.id, dataset.name);
+  if (conflictingName.isDefined())
+    throw Problem.user.datasetNameConflict({ current: conflictingName.get().name, provided: dataset.name });
```

Comment:
Move this above. Situation:

draft "people"
draft "PEOPLE"
publish "people"
API to create "PEOPLE" (should not be allowed)